src/xml.ts

```diff
@@ -209,14 +208,6 @@ export const parseDomFromString = (xml: string): Promise<Document> => {
       return reject(new Error("Not a valid XML document"));
     }
 
-    if (
-      Array.from(dom.childNodes as NodeListOf<Element>).filter(
-        (n) => n.tagName != null && n.childNodes != null
-      ).length !== 1
-    ) {
-      return reject(new Error("Malformed XML; multiple roots detected"));
-    }
-
```

Comment:
@cjbarth I noticed that you already merged another PR which updated xmldom. That is ok, but I noticed that there was added comment https://github.com/node-saml/node-saml/blob/master/src/xml.ts#L217

My opinion is that that is dead code that should be removed.

Reply:
I was thinking about it, then I realized that this code actually protects against a vulnerability and was torn about removing it and then potentially a dependency reintroduces the problem and we are vulnerable again. However, as I type that, I realize they have put in tests to make sure that doesn't happen, so we probably should just remove the old code. I'll create another PR for that.